content/en/blog/_posts/2021-04-22-gateway-api/2021-04-22-gateway-api.md

```diff
@@ -0,0 +1,199 @@
+
+## Getting Hands-on with the Gateway API
+
+In the following example, we’ll demonstrate the relationships between the different API Resources and walk you through a common use case:
+
+*   Team foo has their app deployed in the foo Namespace. They need to control the routing logic for the different pages of their app.
+*   Team bar is running in the bar Namespace. They want to be able to do blue-green rollouts of their application to reduce risk.
+*   The platform team is responsible for managing the load balancer and network security of all the apps in the Kubernetes cluster.
+
+The following foo-route does path matching to various Services in the foo Namespace and also has a default route to a 404 server. This exposes foo-auth and foo-home Services via `foo.example.com/login` and `foo.example.com/home` respectively.:
+
+
+```
+kind: HTTPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: foo-route
+  namespace: foo
+  labels:
+    gateway: external-https-prod
+spec:
+  hostnames:
+  - "foo.example.com"
+  rules:
+  - matches:
+    - path:
+        type: Prefix
+        value: /login
+    forwardTo:
+    - serviceName: foo-auth
+      port: 8080
+  - matches:
+    - path:
+        type: Prefix
+        value: /home
+    forwardTo:
+    - serviceName: foo-home
+      port: 8080
+  - matches:
+    - path:
+        type: Prefix
+        value: /
+    forwardTo:
+    - serviceName: foo-404
+      port: 8080
+```
+
+
+The bar team, operating in the bar Namespace of the same Kubernetes cluster, also wishes to expose their application to the internet, but they also want to control their own canary and blue-green rollouts. The following HTTPRoute is configured for the following behavior:
+
+*   For traffic to `bar.example.com`:
+    *   Send 90% of the traffic to bar-v1
+    *   Send 10% of the traffic to bar-v2
+*   For traffic to `bar.example.com` with the HTTP header `env: canary`:
+    
+    *   Send all the traffic to bar-v2
+    
+![httproute](./httproute.png)
+
+
+
+```
+kind: HTTPRoute
+apiVersion: networking.x-k8s.io/v1alpha1
+metadata:
+  name: bar-route
+  namespace: bar
+  labels:
+    gateway: external-https-prod
+spec:
+  hostnames:
+  - "bar.example.com"
+  rules:
+  - forwardTo:
+    - serviceName: bar-v1
+      port: 8080
+      weight: 90
+    - serviceName: bar-v2
+      port: 8080
+      weight: 10
+  - matches:
+    - headers:
+        values:
+          env: canary
+    forwardTo:
+    - serviceName: bar-v2
+      port: 8080
+```
+
+
+
+### Route and Gateway Binding
+
+So we have two HTTPRoutes matching and routing traffic to different Services. You might be wondering, where are these Routes accessible? Through which networks or IPs are they exposed? 
```

Comment:
nit:

```suggestion
So we have two HTTPRoutes matching and routing traffic to different Services. You might be wondering, where are these Services accessible? Through which networks or IPs are they exposed? 
```

I _think_ users are not looking for a way to access a Route, and that makes less sense than Service, because Routes are a way to represent configuration, while a Service is an abstraction of an application running on a network endpoint. The latter is a bit more concrete.

Reply:
yup this makes sense. Updates